packages/react-router-dom/__tests__/data-browser-router-test.tsx

```diff
@@ -4675,9 +4676,7 @@ function testDomRouter(
           </div>"
         `);
 
-        // Resolve Comp2 loader and complete navigation - Comp1 fetcher is still
-        // reflected here since deleteFetcher doesn't updateState
-        // TODO: Is this expected?
+        // Resolve Comp2 loader and complete navigation
```

Comment:
We had a small bug before where a `deleteFetcher` wouldn't trigger a top-down `state.fetchers` update so `useFetchers` could be slightly delayed in showing the removal until the next top-down update

Reply:
See the new `deleteFetcherAndUpdateState` for the bug fix